Source/WebGPU/WebGPU/Device.mm

```diff
@@ -287,20 +287,16 @@ static void captureFrame(id<MTLDevice> captureObject)
     // https://gpuweb.github.io/gpuweb/#dom-gpudevice-poperrorscope
 
     if (!validatePopErrorScope()) {
-        instance().scheduleWork([callback = WTFMove(callback)]() mutable {
```

Comment:
Ditto for the other patch. This function philosophically should be async; we should be making it properly async rather than temporarily making it sync.

OK for temporary code, but I think we can do better.

Reply:
Why should it be async and doesn't scheduleWork() run on the same dispatch queue which the containing function runs? If so, what is the benefit to making it async?

Reply:
Because that's the API contract of promises, you are guaranteed your promise's `.then` callback won't run in the middle of your `popErrorScope()` call.

Reply:
That's correct but we shouldn't impact the web api for due to how the native API is structured. Instead I think we should ensure the contract on the native API in a way that does not impact the web API.

Reply:
I'm confused. An app that wants to use WebGPU.framework is going to expect that it behaves the same way as the web API does, right?

Reply:
I think so however native WebGPU still seems under development. 

My point is that making the callback asynchronous here negatively impacts users of the web api as now they incur additional latency for the task to return, when it was already asynchronous from the website’s perspective. 

If we want the native API to appear the same, we should implement that in a way that doesn’t add latency to the web api. This patch doesn’t address that as the web api is priority.

Reply:
Is the difference measurable?

Reply:
If the answer is "yes", I propose we add a boolean to `WGPUInstanceCocoaDescriptor` that represents whether violating the asynchronicity guarantees of the spec is more desirable than upholding them, and set the value of the boolean to true in WebKit and false everywhere else, and then implement both codepaths, and use the boolean to select between them.

If the answer is "no" then we should just use the async codepath everywhere.

Reply:
Probably not measurable especially given the discussion is about `popErrorScope`

I will try the new IPC `sendToStreamWithAsyncReply` to see if we can resolve this issue that way. I'm kind of opposed to using NotStreamEncodableReply as that is a workaround

Reply:
@litherum PR updated with `sendWithAsyncReply`, I reverted the changes to `Device.mm`